### PR TITLE
GENIE Rockbox Dirt Workflow

### PR DESCRIPTION
**Pull request by etyley**
This PR adds fcls and filters required to generate dirt interactions using the GENIEHelper `RockBox` configuration, fixing https://github.com/SBNSoftware/sbndcode/issues/176.

The structure of the added code is as follows:
# Gen: 
- Update `genie_sbnd` to contain the rockbox configuration
- Add `prodgenie_rockbox_sbnd` as a base fcl for generating events. This saves any event with either a interaction in the TPC volume or a dirt interaction that deposits `>100 MeV` in the TPC during the beam window
- Add `prodgenie_rockbox_intrnue_sbnd` version of this fcl for NuE only events
- Add `prodoverlay_corsika_cosmics_proton_genie_rockbox_(intrnue_)sbnd.fcl` to add CORSIKA overlay to the GENIE fcls 

# G4:
- Create a new `sbnd_merge_overlay_sim_sources` to manage the merging of the `largeant` instances
- Add `g4_dirt_filter.fcl` to run G4 without `largeant` as that is run in the gen stage of the `rockbox` fcls
- Add `g4_dirt_overlay_filter.fcl` to run `largeant` over CORSIKA and merge the collection with the GENIE `largeant` collection before running the remained of standard G4
- Add `sce` versions of both of the above
- Ideally both workflows would use the same G4 but that is not possible due to ongoing issues with the new larG4: https://cdcvs.fnal.gov/redmine/projects/larsoft/wiki/ReleaseNotes093500#Known-issues. Once this is addressed `largeantCosmic` and the merging should be moved to the `prodoverlay_rockbox` fcls so they can use the `g4_dirt_filter` rather than having their own fcl. 

# Filters:
- Add new `SimEnergyDepFakeTriggerFilter` to utilise the `SimEnergyDeposit` produced by the refactored larG4 to check the amount of energy deposited in the detector during the beam spill time
- Create a new `sbnd_tpc_gennufilter` filter to check if a neutrino interaction occurs inside the TPC volume
- Add `split_tpc_nu` fcl to run over input files and split into two output streams for those with/without a TPC interation. 

As of `v09_35_01` the relevant LArSoft changes have been merged and the necessary addition to `sbnd_data` is done, so this PR is ready for review

### CODE REVIEW

File: sbndcode/Filters/fcls/simenergydepfaketriggerfilter.fcl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+BEGIN_PROLOG
+
+sbnd_simenergydepfaketriggerfilter:
+{
+  module_type:     "SimEnergyDepFakeTriggerFilter"
+
+  # Select the beam spill time to match that in sbnd_filtergenintime
+  BeamTimeMin:    -0.2        # Minimum time of beam window [us]
+  BeamTimeMax:     1.9        # Maximum time of beam window [us]
+  EnergyDeposit:   100        # Minimum energy deposit in TPC for trigger [MeV]
```
Comment (jzennamo) [inline sbndcode/Filters/fcls/simenergydepfaketriggerfilter.fcl:10]: In the future we should think if this is the best value for this, for now I am happy to proceed but we should open an issue to verify this is the optimal energy cut.

**Review by jzennamo — COMMENTED** [review]
This is excellent! One question before I formally approve, has this been tested all the way through reco2 to make sure the full end-to-end workflows work with these events?

File: sbndcode/Filters/fcls/simenergydepfaketriggerfilter.fcl (hunk 1 above)
Comment (etyley) [reply sbndcode/Filters/fcls/simenergydepfaketriggerfilter.fcl:10]: I agree, I will do that once this has been merged.

**Comment by etyley** [comment]
I have tested this end-to-end with an older release and am currently re-running through the validation now that everything has been updated to `v09_35_01`

**Comment by jzennamo** [comment]
awesome, that is sufficient for me, but let us know how the `v09_35_01` test goes!

**Review by jzennamo — APPROVED** [review]

File: sbndcode/JobConfigurations/standard/g4/g4_dirt_overlay_filter.fcl [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+physics.producers.ionandscint.InputModuleLabels: ["largeant"]
+physics.producers.ionandscintout.InputModuleLabels: ["largeant"]
+
+physics.simulate: [ rns , loader, largeantcosmic , largeant , ionandscint , ionandscintout , pdfastsim , pdfastsimout , simdrift , mcreco ]
+
+# Remove the GENIE and CORSIKA specific largeant instances and keep the merged one
+outputs.out1.outputCommands: [ "keep *_*_*_*"
+                             , "drop *_largeantnu_*_*"
+                             , "drop *_largeantcosmic_*_*"
+                             ]
```
Comment (marcodeltutto) [inline sbndcode/JobConfigurations/standard/g4/g4_dirt_overlay_filter.fcl:25]: If PR #206 is approved, do you mind dropping the larg4 SimEnergyDeposits here too? Thanks!

**Review by marcodeltutto — APPROVED** [review]
Thanks very much @etyley! Looks great

**Comment by pgreen135** [comment]
trigger build

File: sbndcode/JobConfigurations/standard/g4/g4_dirt_overlay_filter.fcl (hunk 2 above)
Comment (pgreen135) [reply sbndcode/JobConfigurations/standard/g4/g4_dirt_overlay_filter.fcl:25]: @etyley #206 has now been approved and merged, so you can now update this to also drop these products

**Comment by pgreen135** [comment]
trigger build

**Pull request merged**